src/OpenLoco/src/Map/TileManager.cpp

```diff
@@ -1267,6 +1268,86 @@ namespace OpenLoco::World::TileManager
         return totalCost;
     }
 
+    // 0x004C4C28
+    uint32_t adjustWaterHeight(World::Pos2 pos, SmallZ targetHeight, std::set<World::Pos3, LessThanPos3>& removedBuildings, uint8_t flags)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Construction);
+        GameCommands::setPosition(World::Pos3(pos.x + World::kTileSize / 2, pos.y + World::kTileSize / 2, targetHeight * kMicroToSmallZStep));
+
+        if (targetHeight < 4)
+        {
+            GameCommands::setErrorText(StringIds::error_too_low);
+            return GameCommands::FAILURE;
+        }
+
+        if (targetHeight > 116)
+        {
+            GameCommands::setErrorText(StringIds::error_too_high);
+            return GameCommands::FAILURE;
+        }
+
+        currency32_t totalCost = 0;
+
+        if (flags & GameCommands::Flags::apply)
+        {
+            removeSurfaceIndustry(pos);
+
+            if (!isEditorMode())
+            {
+                setTerrainStyleAsCleared(pos);
+            }
+
+            auto clearHeight = getHeight(pos).landHeight;
+            removeAllWallsOnTile(toTileSpace(pos), clearHeight / 4);
+        }
+
+        auto tile = get(pos);
+        auto* surface = tile.surface();
+
+        auto referenceZ = surface->baseZ();
+        auto water = surface->water();
+        if (water > 0)
+        {
+            referenceZ = surface->waterHeight();
+        }
+
+        if (targetHeight < referenceZ)
```

Comment:
referenceZ and targetHeight are used as the base and clear heights. The smaller of the two is the base and the larger is the clear.

Reply:
so you need to do:
```
const auto baseTargetZ = std::min(targetHeight, referenceZ);
const auto clearTargetZ = std::max(targetHeight, referenceZ);
```
And then use the correct ones for the next part of the code